subprojects/packagefiles/tree-sitter/meson.build

```
@@ -0,0 +1,28 @@
+project('tree-sitter', 'c'
+  version: '0.15.7',
+  license: 'MIT',
+)
+
+tree_sitter = library('tree-sitter', [
+  'lib/src/lib.c',
+  ],
+  include_directories: [
+    include_directories('lib/src'),
+    include_directories('lib/include'),
+    include_directories('lib/utf8proc')
+  ],
+)
+
+tree_sitter_dep = declare_dependency(
+  link_with: tree_sitter,
+  include_directories: include_directories('lib/include'),
+)
+
+
+pkg = import('pkgconfig')
+pkg.generate(libraries : tree_sitter,
```

```suggestion
pkg.generate(tree_sitter,
```

See the pkgconfig module documentation:
> Since 0.46 a StaticLibrary or SharedLibrary object can optionally be passed as first positional argument. If one is provided a default value will be provided for all required fields of the pc file:

---

You will need to then add `meson_version: '>=0.46.0',` to the project() declaration.